### PR TITLE
Add docs build to Jenkinsfile

### PR DESCRIPTION
**Pull request by ruflin**
Remove docs build from Travis. This reduces the number of builds needed on Travis so we can have Travis for the things we do not have on Jenkins more effective.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -101,6 +101,101 @@ pipeline {
...
+              unstash 'source'
+              dir("${BASE_DIR}"){
+                sh """#!/bin/bash
+                set -euxo pipefail
+                make -C  metricbeat docs
+                """
+              }
+            }
+          }
+        }
```
Comment (jsoriano) [inline Jenkinsfile]: Could we create some kind of shared parametrized stages for beats?

File: Jenkinsfile [hunk 2]
```diff
@@ -101,6 +101,101 @@ pipeline {
...
+
+        stage('Metricbeat docs') {
+          agent { label 'linux && immutable' }
+          options { skipDefaultCheckout() }
+          environment {
+            PATH = "${env.PATH}:${env.WORKSPACE}/bin"
+            HOME = "${env.WORKSPACE}"
+            GOPATH = "${env.WORKSPACE}"
+          }
```
Comment (jsoriano) [inline Jenkinsfile]: Is there some way to set environment variables for all stages? We probably want these variables to be set in all beats builds.

**Comment by ruflin** [comment]
@kuisathaverat Could you have a look at the questions above from @jsoriano ?

File: Jenkinsfile (hunk 2 above)
Comment (kuisathaverat) [reply Jenkinsfile]: WORKSPACE environment variable depends on the Agent where the job is running, I mean, when the Agent is launch this env var is set, it is true that we have one shot agents and probably the value is the same always but you can not ensure it, so the correct way to define those variables is in the environment section of the stage that launch the Agent, however, you can define them to a fixed value in the root environment section, I do not recommend it.

```
pipeline {
  environment {
            AGENT_WORKSPACE="/var/lib/jenkins/workspace/JOBNAME"
            PATH = "/bin:/sbin:/usr/bin:/usr/sbin:${env.AGENT_WORKSPACE}/bin"
            HOME = "${env.AGENT_WORKSPACE}"
            GOPATH = "${env.AGENT_WORKSPACE}"
  }  
```

Comment (kuisathaverat) [reply Jenkinsfile]: if you do not want to repeat it over and over again you could define a function like the following and call it as first step on each steps section

```
def prepareEnvironment(){
            env.PATH = "${env.PATH}:${env.WORKSPACE}/bin"
            env.HOME = "${env.WORKSPACE}"
            env.GOPATH = "${env.WORKSPACE}"
}
```

Comment (jsoriano) [reply Jenkinsfile]: Thanks, I think that defining a function for these common environment variables can make sense here.

**Comment by ruflin** [comment]
Closing this at the moment as I don't have the time to finish it, will potentially reopen again later. Thanks for all the comments, this will become useful.

**Pull request closed without merging**